completions/ssh-keyscan

```diff
@@ -0,0 +1,39 @@
+# ssh-keyscan(1) completion                                -*- shell-script -*-
+
+_comp_cmd_ssh_keyscan()
+{
+    local cur prev words cword
+    _init_completion || return
+
+    local ipvx
+
+    case $prev in
+        -*4*)
+            ipvx=-4
+            ;;
+        -*6*)
+            ipvx=-6
+            ;;
+        -*f)
+            _filedir
+            return
+            ;;
+        -*p | -*T)
+            return
+            ;;
+        -*t)
+            COMPREPLY=($(compgen -W 'dsa ecdsa ed25519 rsa' -- "$cur"))
```

Comment:
I'm not familiar with `ssh-keyscan`, but `man ssh-keyscan` says (boldfaced by me)

> - -t type
>   Specify the type of the key to fetch from the scanned hosts.  The possible values are “dsa”, “ecdsa”, “ed25519”, or “rsa”.  **Multiple values may be specified by separating them with commas.**  The default is to fetch “rsa”, “ecdsa”, and “ed25519” keys.

Maybe something like the following?

```suggestion
            local prefix=
            [[ $cur == *,* ]] &&
                local prefix=${cur%,*}, cur=${cur%%*,}
            COMPREPLY=($(compgen ${prefix:+-P "$prefix"} -W 'dsa ecdsa ed25519 rsa' -- "$cur"))
```

Reply:
Ah, good catch! Implemented in 1893481a482c6d6088149d55a1930b4df8252e8b using `_comp_delimited`, it's nicer because it shows only the currently being completed value in its suggestions instead of the whole chain, and removes values already present.

Reply:
OK! Thank you! I didn't know this useful utility `_comp_delimited`!